update solid mechanics, scripts, cross_product, schemes, models..

josep-m-carbonell opened the pull request.

lluis-mv committed:
Merge remote-tracking branch 'origin/feature-gpfem-constitutive-develop' into feature-umat-models
Rebase the branch

lluis-mv committed:
Merge remote-tracking branch 'origin/master' into feature-umat-models

lluis-mv committed:
there is a mismatch in LINE_LOAD_VECTOR

(sort of solves the problem for me now)

lluis-mv committed:
the pseudo-tests of PFEM/SMA now work

Remove mesh_id, update MainKratos.py and remove stupid files

lluis-mv committed:
modifications on the scheme + pseudoTest

Scheme *Add capability for UWwP Element *Modifications due to the Dirichlet BC and a sort of test

lluis-mv committed:
Merge remote-tracking branch 'origin/master' into gpfem/feature-full-biot-element

lluis-mv committed:
Merge remote-tracking branch 'origin/master' into gpfem/feature-umat-models

lluis-mv committed:
Cmake modification

something about API

lluis-mv committed:
continuation

Removing unnecessary code (including laws), modifications on the Serializer, understand hypoplastic model for clays,...

lluis-mv committed:
Comment some that created a warning

The computation of the hardening was not correctly coded (some sort of segmentation fault was produced). For now, only J2 without hardening until I understand the implementation

mpentek committed:
Add files via upload

Committing the changes with recommended edits to the ALE
remedied bugs
updated formatting for solver naming
added more explicit error message

RiccardoRossi committed:
Merge pull request #1001 from KratosMultiphysics/hotfix-correcting-interval-defaults

defaults were wrongly set

mpentek committed:
Removed unnecessary "raise"

after else in line 25 -> removed old raise and output:  raise Exception("the requested solver type is not in the python solvers wrapper")

roigcarlo committed:
Merge pull request #1003 from KratosMultiphysics/release/ale-scripts

Ale scripts

lluis-mv committed:
Element UWwP. First steps

modification of the scheme to update scalar variables, first push of the element, ...
It compiles but it does not work (in the sence that it has not been tested).

josep-m-carbonell committed:
3D constitutive tensor to 2D component correction

josep-m-carbonell committed:
Merge branch 'solid/feature-correct-constitutive-transformation' into solid/develop

josep-m-carbonell committed:
Merge branch 'pfem/feature-check-dimension-for-volume' into solid/develop

josep-m-carbonell committed:
unitialized value

josep-m-carbonell committed:
Merge branch 'solid/feature-correct-constitutive-transformation' into solid/develop

josep-m-carbonell committed:
Merge branch 'master' into solid/develop

josep-m-carbonell committed:
constitutive models corrections

josep-m-carbonell committed:
constitutive models corrections

josep-m-carbonell committed:
round time correction

josep-m-carbonell committed:
round time correction

roigcarlo committed:
Merge branch 'Release-5.2' into dem-bugfix-reactions

roigcarlo committed:
Merge pull request #1018 from KratosMultiphysics/dem-bugfix-reactions

Dem bugfix reactions

loumalouomega committed:
Hotfix BODYFORCE

Solves #1019

loumalouomega committed:
Fix

lluis-mv committed:
Merge remote-tracking branch 'origin/master' into gpfem/feature-full-biot-element

josep-m-carbonell committed:
Merge pull request #1013 from KratosMultiphysics/feature-hotfix-constitutive-utility

3D constitutive tensor to 2D component correction

lluis-mv committed:
it works in the "small strain regime"

change water_pressure_acceleration to water_pressure_accelerationN (due to something with the legacy structural variables??)[more work is required]
Add a missing matrix in the stiffness matrix.
No valgrind error, hehe

lluis-mv committed:
Pseudo test for the UWwP (semi-full-biot)

It seems to work!!

josep-m-carbonell committed:
Merge branch 'master' into solid/develop

lluis-mv committed:
Add Stabilization Term

now it works perfectly (almost) :'P

lluis-mv committed:
Add Element to problemtype

:

lluis-mv committed:
Merge remote-tracking branch 'origin/master' into gpfem/feature-full-biot-element

lluis-mv committed:
I messed up 

correction of the modification on the previous problem type.... extra endif

lluis-mv committed:
Merge remote-tracking branch 'origin/master' into gpfem/feature-umat-models

josep-m-carbonell committed:
checking elastic model

josep-m-carbonell committed:
adding dynamic mass matrix options

josep-m-carbonell committed:
adding some noalias...

josep-m-carbonell committed:
restore needed clears

josep-m-carbonell committed:
output print modified

josep-m-carbonell committed:
add flush in application importer

to write in gid info window correctly

josep-m-carbonell committed:
fixes python scripts, writing in GiD tcl terminal and Clws

loumalouomega committed:
Merge pull request #1022 from KratosMultiphysics/hotfix-bodyforce-Release-5.2

Hotfix bodyforce release 5.2

lluis-mv committed:
test in SS regime

reduction to the small strain regime

lluis-mv committed:
changes in the JSON of problemtype

I created an error, now solved

lluis-mv committed:
Corrections and register SSUWwP element

lluis-mv committed:
Merge remote-tracking branch 'origin/master' into gpfem/feature-full-biot-element

lluis-mv committed:
(remove one compiler warning due to CrossProduct)

josep-m-carbonell committed:
move python registers to core for variables declared in core

josep-m-carbonell committed:
move python registers to core for variables declared in core

josep-m-carbonell committed:
undo sys.stdout.flush and correct scripts

josep-m-carbonell committed:
undo sys.stdout.flush and correct scripts

josep-m-carbonell committed:
Merge pull request #1036 from KratosMultiphysics/solid/Fixes-Release-5.2

first fixes in solid and constituitve apps after first tests with the release 5.2

josep-m-carbonell committed:
Merge branch 'Release-5.2' into solid/Fixes-Release-5.2

josep-m-carbonell committed:
Merge branch 'solid/Fixes-Release-5.2' into solid/develop

mpentek committed:
Removed a duplicate error raise

mpentek committed:
Merge pull request #1050 from KratosMultiphysics/mpentek-patch-2

Removed a duplicate error raise

loumalouomega committed:
Add files via upload

lluis-mv committed:
Add first draft of UJWwP

it compiles and seems to work :'P

lluis-mv committed:
pseudo-test

UJwP

lluis-mv committed:
modify the scheme

¿not sure way?

lluis-mv committed:
Ups, I don't want to be virtual of that

lluis-mv committed:
Merge remote-tracking branch 'origin/master' into gpfem/feature-full-biot-element

josep-m-carbonell committed:
adding current_value as string field

lluis-mv committed:
I do not understand

josep-m-carbonell committed:
Merge branch 'Release-5.2' into solid/develop

josep-m-carbonell committed:
Merge branch 'master' into solid/develop

josep-m-carbonell committed:
Merge branch 'Release-5.2' into solid/Fixes-Release-5.2

lluis-mv committed:
Typo in FixDerivedVariable

lluis-mv committed:
add numpy in assign. It adds more features

lluis-mv committed:
merge

lluis-mv committed:
Merge remote-tracking branch 'origin/master' into gpfem/feature-umat-models

josep-m-carbonell committed:
Merge remote-tracking branch 'kratos/solid/develop' into solid/develop

loumalouomega committed:
Update kratos_main_structural.py

loumalouomega committed:
Updating model.h and model.cpp

Updates from @RiccardoRossi

josep-m-carbonell committed:
adding ballast coefficient

josep-m-carbonell committed:
corrections in elastic strings

lluis-mv committed:
Correct Typo in the Fixing of Dynamic variables

lluis-mv committed:
Correct Typo in the Fixing of Dynamic variables

josep-m-carbonell committed:
updating cross product first attempt

lluis-mv committed:
Add problem path in material import (it was missing) correct 1 crossProd

The cross product only produced a warning in my computer (since the file is not in the CMake)

lluis-mv committed:
Add problem path in material import (it was missing) correct 1 crossProd

The cross product only produced a warning in my computer (since the file is not in the CMake)

josep-m-carbonell committed:
Merge branch 'master' into solid/develop

lluis-mv committed:
Merge branch 'gpfem/feature_contact_thread_safe' into gpfem/feature-umat-models

josep-m-carbonell committed:
fix M_PI in a test

josep-m-carbonell committed:
Merge branch 'solid/Fixes-Release-5.2' into solid/develop

josep-m-carbonell committed:
start a class to split time integration of variables

lluis-mv committed:
Change Tolerance in EigenVectors Calc (EECC Old)

It didn't work with very small displacements.

lluis-mv committed:
Merge remote-tracking branch 'origin/master' into gpfem/feature-umat-models

roigcarlo committed:
Merge pull request #1096 from KratosMultiphysics/release/model-main-script

Release/model main script

lluis-mv committed:
Merge branch 'gpfem/feature-umat-models' into gpfem/feature-finish-at-least-one-EECC

josep-m-carbonell committed:
correction to accept shells plane stress dimension check

josep-m-carbonell committed:
correction to accept shells plane stress dimension check

josep-m-carbonell committed:
working on new schemes

josep-m-carbonell committed:
Merge branch 'Release-5.2' into solid/Fixes-Release-5.2

josep-m-carbonell committed:
undo new cross product

lluis-mv committed:
add missing condition in Sloan algorithm

lluis-mv committed:
Define CriticalState parameters in mat_variables

remove them from PfemSolidMec, then the MCC of ConstitutiveModels may get them

lluis-mv committed:
small correction in one element (UJElement)

to make it also "compatible" with the new EECC

lluis-mv committed:
some sort of test. Check numerical against analytical and published

It looks great!

lluis-mv committed:
changes in the check of constit models

Ask JMC about it

lluis-mv committed:
Add cam clay small test

I think the idea is good, but...

lluis-mv committed:
cosmetic changes

lluis-mv committed:
modify CMA test file to make it run

josep-m-carbonell committed:
Merge branch 'solid/Fixes-Release-5.2' into solid/develop

josep-m-carbonell committed:
Merge branch 'master' into solid/develop

josep-m-carbonell committed:
scheme update developments

josep-m-carbonell committed:
Merge branch 'master' into solid/develop

lluis-mv committed:
Merge remote-tracking branch 'origin/master' into gpfem/feature-finish-at-least-one-EECC

lluis-mv committed:
beggining with the LodeAngle thing

lluis-mv committed:
modifications constitutive laws.

josep-m-carbonell committed:
start new schemes update

josep-m-carbonell committed:
update cross products

josep-m-carbonell committed:
update cross products

josep-m-carbonell committed:
update pointer register

josep-m-carbonell committed:
solving integration method python register

josep-m-carbonell committed:
updating python scripts

josep-m-carbonell committed:
update bossak schemes

josep-m-carbonell committed:
print updates and segfault corrected

josep-m-carbonell committed:
Merge branch 'gpfem/feature-finish-at-least-one-EECC' into solid/develop

josep-m-carbonell committed:
modify some overwrites

two unneeded files deleted

josep-m-carbonell committed:
modify some overwrites

two unneeded files deleted

josep-m-carbonell committed:
Merge branch 'gpfem/feature-remove-code' into solid/develop

josep-m-carbonell committed:
Merge branch 'master' into solid/develop

josep-m-carbonell committed:
Merge branch 'solid/develop' into solid/update_integration_schemes

josep-m-carbonell committed:
Merge branch 'master' into solid/update_integration_schemes

josep-m-carbonell committed:
backup file removed

josep-m-carbonell committed:
Merge branch 'master' into solid/update_integration_schemes

josep-m-carbonell committed:
windows compilation warnings solved

josep-m-carbonell committed:
Merge branch 'master' into solid/update_integration_schemes

josep-m-carbonell committed:
start cleanina tests for compatibility

josep-m-carbonell committed:
update in model file name supply

josep-m-carbonell committed:
update tests solid app

josep-m-carbonell committed:
Merge branch 'master' into solid/update_integration_schemes

josep-m-carbonell committed:
delete unneded check parameter

josep-m-carbonell committed:
adding proper constructors to applications

josep-m-carbonell committed:
Merge branch 'feature-app-constructor' into solid/update_integration_schemes

josep-m-carbonell committed:
Merge branch 'master' into solid/update_integration_schemes

josep-m-carbonell committed:
clean PfemSolid app and update scripts

josep-m-carbonell committed:
Merge branch 'master' into solid/update_integration_schemes

josep-m-carbonell committed:
avoid importing numpy

problems with GiDinterface

josep-m-carbonell committed:
correct seg-fault

still not obtaining the same results

josep-m-carbonell committed:
update damage models

josep-m-carbonell committed:
Merge branch 'master' into solid/update_integration_schemes

josep-m-carbonell committed:
legacy constitutive CalculateValue

roigcarlo commented on kratos/tests/geometries/test_quadrilateral_3d_4.cpp:
```diff
@@ -19,6 +19,7 @@
 #include "testing/testing.h"
 #include "geometries/quadrilateral_3d_4.h"
 #include "tests/geometries/test_geometry.h"
+#include "includes/global_variables.h"
```
Why has this been included?

josep-m-carbonell replied:
I don't know about this change. I can not remember if I made it, but I guess I don't. It is possible that this line was there before and it was cleaned but in some merge remained there in my local branch. However it is strange. If it is not needed it must be taken away.

RiccardoRossi replied:
@josep-m-carbonell are u still using GitKraken? i am asking since we are reaching the conclusion that it is too easy to make mistakes when using it, since it brings to one click "strange" operations.

roigcarlo commented on applications/PFEMapplication/CMakeLists.txt~:
```diff
@@ -1,46 +0,0 @@
-# message("KratosIncompressibleFluidApplication subdir inc_dirs = ${inc_dirs}")
-
-# Add to the KratosMultiphisics Python module
-install(FILES "${CMAKE_CURRENT_SOURCE_DIR}/PFEMApplication.py" DESTINATION KratosMultiphysics )
```
This cannot be deleted...

roigcarlo replied:
Forget about this, i didn't saw the ~

josep-m-carbonell replied:
Yes this application was deleted some time ago.

josep-m-carbonell replied:
@roigcarlo  it stills blocks the pr

RiccardoRossi replied:
I am not involved in this review, so i will not say anything about the code.
However as a comment, it would be very good if you avoid this sort of enormous PRs.
A PR should include whenever possible only a few files or specific modifications. In this one you are merging a lot of different PRs into your own branch prior to merging it to the master. This makes the reviewing much more difficult and slower

josep-m-carbonell committed:
correction global vars

lluis-mv committed:
Remove half-done code

lluis-mv committed:
Borja Hyperelastic model. minor modification

I was counting on the rStressMatrix.clear() of the CalculateStressTensor before calling CalculateAndAddStressTensor

lluis-mv committed:
few modifications in PfemSolid script

lluis-mv reviewed (dismissed):
I directly made some modifications (modified cam clay and the script to write the json in PfemSolid problemtype).
In general it is OK with me.

josep-m-carbonell committed:
update tests

josep-m-carbonell committed:
Merge remote-tracking branch 'kratos/solid/update_integration_schemes' into solid/update_integration_schemes

josep-m-carbonell committed:
corrections for the shell tests

lluis-mv committed:
ZeroMatrix(m,n)

Reset the code as it was

pooyan-dadvand commented:
@josep-m-carbonell I would strongly suggest you to avoid such a huge PR and even worse before the beginning of the holidays.
